staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -529,21 +538,17 @@ func (p *concurrentOrderedEventProcessing) collectEventProcessing(ctx context.Co
 		select {
 		case <-ctx.Done():
 			return
-		case e = <-processingResponse:
+		case r = <-processingResponse:
 		}
-		if e == nil {
-			continue
+		if r.err != nil {
+			p.wc.sendError(r.err)
+			return
 		}
-		if len(p.output) == cap(p.output) {
-			klog.V(3).InfoS("Fast watcher, slow processing. Probably caused by slow dispatching events to watchers", "outgoingEvents", outgoingBufSize, "objectType", p.objectType, "groupResource", p.groupResource)
+		if r.event == nil {
+			continue
 		}
-		// If user couldn't receive results fast enough, we also block incoming events from watcher.
-		// Because storing events in local will cause more memory usage.
-		// The worst case would be closing the fast watcher.
-		select {
-		case <-ctx.Done():
```

Comment:
maybe avoid this refactor in the version we're backporting?

Reply:
done